orca-uv.py

```diff
@@ -0,0 +1,417 @@
+    if save_spectrum:
+        filename, file_extension = os.path.splitext(path)
+
+        if nm_plot:
+            # plt.xlim(60,500)
+            plt.ylim(0.0, plt_y_lim)
+            plt.savefig(f"{filename}/abs_spectrum_nm.png", dpi=figure_dpi)
+        else:
+            # plt.xlim(2.50,15)
+            plt.savefig(f"{filename}/abs_spectrum_eV.png", dpi=figure_dpi)
+
+    # export data
+    if export_spectrum:
+        # get data from plot (window)
+        plotdata = ax.lines[0]
+        xdata = plotdata.get_xdata()
+        ydata = plt_range_gauss_sum_y
+        xlimits = plt.gca().get_xlim()
+        try:
+            spectrum_file_without_extension = os.path.splitext(spectrum_file)[0]
+            with open(
+                spectrum_file_without_extension + "-smooth.DAT", "w"
+            ) as output_file:
+                for elements in range(len(xdata)):
+                    if xlimits[0] <= xdata[elements] <= xlimits[1]:
+                        output_file.write(
+                            str(xdata[elements])
+                            + export_delim
+                            + str(ydata[elements])
+                            + "\n"
+                        )
+        # file not found -> exit here
+        except IOError:
+            print("Write error. Exit.", flush=True)
+        except Exception as e:
+            print("Rank: ", comm_rank, " encountered Exception: ", e.message, e.args)
+
+    # show the plot
+    if show_spectrum:
+        plt.show()
+
+    plt.close(fig)
+
+
+def smooth_spectra(comm, path, min_energy, max_energy, min_wavelength, max_wavelength):
+    comm.Barrier()
+    comm_size = comm.Get_size()
+    comm_rank = comm.Get_rank()
+
+    if comm_rank == 0:
+        print("=" * 50, flush=True)
+        print("Smooth spectra", flush=True)
+        print("=" * 50, flush=True)
+    comm.Barrier()
+    dirs = None
+    if comm_rank == 0:
+        dirs = [f.name for f in os.scandir(path) if f.is_dir()]
+
+    dirs = comm.bcast(dirs, root=0)
+    rx = list(nsplit(range(len(dirs)), comm_size))[comm_rank]
+    total = rx.stop - rx.start
+    count = 0
+    for dir in sorted(dirs)[rx.start : rx.stop]:
+        count = count + 1
+        print(
+            "s Rank: ",
+            comm_rank,
+            " - dir: ",
+            dir,
+            ", remaining: ",
+            total - count,
+            flush=True,
+        )
+        # collect information about molecular structure and chemical composition
+        if os.path.exists(path + "/" + dir + "/" + "orca.stdout"):
+            smooth_spectrum(
+                path, dir, min_energy, max_energy, min_wavelength, max_wavelength
+            )
+
+
+if __name__ == "__main__":
+    path = "/Users/7ml/Documents/SurrogateProject/ElectronicExcitation/GDB-9-Ex-ORCA-EOM-CCSD"
+    min_energy = 0.0
+    max_energy = 100.0
+    min_wavelength = 0.0
+    max_wavelength = 750.0
+
+    communicator = MPI.COMM_WORLD
+    draw_2Dmols(communicator, path, save_moldraw)
```

Comment:
Whats `save_moldraw`? I guess its a boolean value. Also, should you call `draw_2Dmol` instead of `draw_2Dmols`?

Reply:
`draw_2Dmol**s**` iterates over all the molecular directories assigned to an MPI process, and calls `draw_2Dmol` on each directory.